addons/GDQuest_GDScript_formatter/plugin.gd

```diff
@@ -403,6 +471,105 @@ func format_code(script: GDScript, force_reorder := false) -> String:
 		return ""
 
 
+## Lints a GDScript file using the GDScript Formatter's linter,
+## and returns an array of lint issues.
+func lint_code(script: GDScript) -> Array:
+	var script_path := script.resource_path
+	var output: Array = []
+	var formatter_arguments: Array = ["lint", ProjectSettings.globalize_path(script_path)]
+
+	var exit_code := OS.execute(get_editor_setting(SETTING_FORMATTER_PATH), formatter_arguments, output)
+	if exit_code == OK:
+		return [] # No issues found
+
+	if exit_code == 1:
+		# Parse lint output - the output is a single string with multiple lines
+		var issues = []
+		for output_item in output:
+			var lines = output_item.split("\n")
+			for line in lines:
+				var trimmed_line = line.strip_edges()
+				if trimmed_line.is_empty():
+					continue
+				var issue = parse_lint_issue(trimmed_line)
+				if issue != null and not issue.is_empty():
+					issues.push_back(issue)
+		return issues
+
+	push_error("Lint GDScript failed: " + script_path)
+	push_error("\tExit code: " + str(exit_code) + " Output: " + (output.front().strip_edges() if output.size() > 0 else "No output"))
+	return []
+
+
+## Parses a lint issue line and returns a dictionary with issue information
+func parse_lint_issue(line: String) -> Dictionary:
+	# Expected format: filename:line:rule:severity: message
+	var parts = line.split(":", 4)
+	if parts.size() < 5:
+		return { }
+
+	return {
+		"line": int(parts[1]) - 1, # Convert to 0-based indexing
+		"rule": parts[2],
+		"severity": parts[3],
+		"message": parts[4].strip_edges(),
+	}
+
+
+## Applies lint highlighting to the code editor
+func apply_lint_highlights(code_edit: CodeEdit, issues: Array) -> void:
+	clear_lint_highlights(code_edit)
+
+	for issue in issues:
+		var line_number: int = issue.line
+		var severity: String = issue.severity
+		var message: String = issue.message
+
+		# Set line background color based on severity
+		var color: Color
+		if severity == "error":
+			color = Color(1, 0, 0, 0.1)
+		else: # warning
+			color = Color(1, 1, 0, 0.1)
+
+		code_edit.set_line_background_color(line_number, color)
+
+		# Add gutter icon for severity
+		var gutter_index = LINT_ICON_GUTTER
+		var icon_name = "StatusError" if severity == "error" else "StatusWarning"
+		var icon = EditorInterface.get_editor_theme().get_icon(icon_name, "EditorIcons")
+		code_edit.set_gutter_type(LINT_ICON_GUTTER, CodeEdit.GutterType.GUTTER_TYPE_ICON)
+		code_edit.set_line_gutter_icon(line_number, LINT_ICON_GUTTER, icon)
```

Comment:
There _is_ a way you can use `GUTTER_TYPE_TEXT` with `set_line_gutter_text` but I'm not sure what the use case there is - the text just goes into the editor itself.

I couldn't find a way to add text to the gutter icon when you hover over it, so I resorted to just printing out the lint issues to the editor's output.

Reply:
> Godot itself also outputs warnings and things like these to a separate folded panel

Yeah, I was trying to find a way to insert comments into that but unfortunately it's not exposed in the `EditorInterface` or `CodeEdit` anywhere. Could be a nice proposal to make 🤔